mesh-worker-service/src/main/java/io/functionmesh/compute/models/MeshWorkerServiceCustomConfig.java

```diff
@@ -0,0 +1,140 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.functionmesh.compute.models;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.functionmesh.compute.functions.models.V1alpha1FunctionSpecPodVolumeMounts;
+import io.functionmesh.compute.functions.models.V1alpha1FunctionSpecPodVolumes;
+import io.functionmesh.compute.sinks.models.V1alpha1SinkSpecPodVolumeMounts;
+import io.functionmesh.compute.sinks.models.V1alpha1SinkSpecPodVolumes;
+import io.functionmesh.compute.sources.models.V1alpha1SourceSpecPodVolumeMounts;
+import io.functionmesh.compute.sources.models.V1alpha1SourceSpecPodVolumes;
+import io.kubernetes.client.openapi.models.V1Volume;
+import io.kubernetes.client.openapi.models.V1VolumeMount;
+import lombok.Data;
+import lombok.experimental.Accessors;
+import org.apache.pulsar.common.configuration.FieldContext;
+import org.apache.pulsar.common.util.ObjectMapperFactory;
+
+import java.util.List;
+import java.util.Map;
+
+@Data
+@Accessors(chain = true)
+public class MeshWorkerServiceCustomConfig {
+    @FieldContext(
+            doc = "Enable user to upload custom function/source/sink jar/nar"
+    )
+    protected boolean uploadEnabled = false;
+
+    @FieldContext(
+            doc = "Enable the function api endpoint"
+    )
+    protected boolean functionEnabled = true;
+
+    @FieldContext(
+            doc = "Enable the sink api endpoint"
+    )
+    protected boolean sinkEnabled = true;
+
+    @FieldContext(
+            doc = "Enable the source api endpoint"
+    )
+    protected boolean sourceEnabled = true;
+
+    @FieldContext(
+            doc = "the directory for dropping extra function dependencies. "
+    )
+    protected String extraDependenciesDir = "/pulsar/lib/";
+
+    @FieldContext(
+            doc = "VolumeMount describes a mounting of a Volume within a container."
+    )
+    protected List<V1VolumeMount> volumeMounts;
+
+    @FieldContext(
+            doc = "List of volumes that can be mounted by containers belonging to the function/connector pod."
+    )
+    protected List<V1Volume> volumes;
+
+    @FieldContext(
+            doc = "ownerReference"
+    )
+    protected Map<String, Object> ownerReference;
+
+    @FieldContext(
+            doc = "if allow user to change the service account name with custom-runtime-options"
+    )
+    protected boolean allowUserDefinedServiceAccountName = true;
```

Comment:
I think we should set the default value to `false` otherwise it potentially exposes vulnerabilities to those clusters deployed without aware of this configuration.